core/discovery/interface.go

```diff
@@ -28,3 +28,9 @@ type ProposalRegistry interface {
 	PingProposal(proposal market.ServiceProposal, signer identity.Signer) error
 	UnregisterProposal(proposal market.ServiceProposal, signer identity.Signer) error
 }
+
+// ProposalFinder allows to search proposals by specified params
+type ProposalFinder interface {
```

Comment:
still not a fan of defining interfaces nowhere close to where they are being used.

Reply:
Yes, me too, just this interface is used in several places and don't know how to solve that.
Don't want to add ti to `di.go` either, because then everybody includes `cmd` package